tests/ludwig/datasets/model_configs/train_all_model_configs.py

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+#
+# Trains a ludwig model for every dataset which has a default_model_config.
+# You must have valid kaggle credentials in your environment, a few GB of disk space, and good internet bandwidth.
+# Also, for each dataset associated with a Kaggle competition you'll need to sign in to Kaggle and accept the terms of
+# the competition.
+#
+import multiprocessing
+import time
+from dataclasses import dataclass
+from typing import Optional
+
+import pandas as pd
+
+from ludwig import datasets, visualize
+from ludwig.api import LudwigModel
+
+
+@dataclass
+class TrainingResults:
+    """Results of a training run for a dataset."""
+
+    dataset_version: str
+    dataset_name: str
+    has_config: bool
+    output_directory: Optional[str] = None
+    splits: Optional[str] = None
+    metric: Optional[str] = None
+    performance: Optional[float] = None
+    load_time: Optional[float] = None
+    train_time: Optional[float] = None
+    eval_time: Optional[float] = None
+
+
+def _train_dataset_process(dataset_name, results_queue):
+    """Runs each train job in a new process."""
+    load_start_time = time.time()
+    dataset = datasets.get_dataset(dataset_name)
+    config = dataset.default_model_config
+    df = dataset.load()
+    load_end_time = time.time()
+    if "split" not in df:
+        df["split"] = 0
```

Comment:
As splits might change based on the dataset config, should we consider using `from ludwig.utils.dataset_utils import get_repeatable_train_val_test_split`

Reply:
Some of the datasets here have splits which match the splits used in the literature, so I don't want to re-split those.

Also, this test script is more of a sanity check than a benchmark, so I think its OK that some datasets don't have proper splits.